prescient.el

```diff
@@ -714,8 +717,277 @@ Return the sorted list. The original is modified destructively."
+  "Sort fully matched strings in CANDIDATES before other candidates.
+
+REGEXPS are the regexps prescient.el used to filter the candidates.
+IGNORE-CASE is whether case was ignored when filtering."
+  (save-match-data
+    (cl-loop
+     with prioritized-candidates = nil
+     and remaining-candidates = nil
+     and case-fold-search = ignore-case
+     for cand in candidates
+     ;; By this point, after filtering, we know that a candidate
+     ;; matches all regexps, so we don't have to check that.  So long
+     ;; as a candidate is fully matched by one regexp, we move it to
+     ;; the front of the list.
+     if (cl-loop for regexp in regexps
+                 thereis (and (string-match regexp cand)
+                              ;; We can't use `match-beginning' and
+                              ;; `match-end' here, since they seem to
+                              ;; only work on the buffer.
+                              (eq (length cand)
+                                  (length (match-string 0 cand)))))
+     do (push cand prioritized-candidates)
+     else do (push cand remaining-candidates)
+     finally return (nconc (nreverse prioritized-candidates)
+                           (nreverse remaining-candidates)))))
+
+(cl-defun prescient--post-sort (candidates)
+  "Sort CANDIDATES after running filtering.
+
+This function checks for the properties `prescient-regexps' and
+`prescient-ignore-case' on the first candidate in CANDIDATES.
+These properties are set by
+`prescient--completion-style-filter' after filtering, and
+are used for implementing `prescient-sort-full-matches-first'."
+  (let ((regexps (get-text-property 0 'prescient-regexps
+                                    (car candidates)))
+        (ignore-case (get-text-property 0 'prescient-ignore-case
+                                        (car candidates)))
+        (sorted (prescient-sort candidates)))
+    (if (and prescient-sort-full-matches-first regexps)
+        (prescient--sort-full-matches-first sorted regexps ignore-case)
+      sorted)))
+
+(defun prescient--completion-modify-sort (metadata)
+  "Modify METADATA to use prescient.el for sorting if no order given."
+  ;; Based on `completion--flex-adjust-metadata'.
+  ;; Unlike `completion--flex-adjust-metadata', we want to modify
+  ;; sorting even when there is no input.
+  (if (or (null prescient-completion-enable-sort)
+          ;; If there is any existing sorting information, we assume
+          ;; that it is meaningful and that we shouldn't even move the
+          ;; fully matched candidates.
+          (completion-metadata-get metadata 'display-sort-function)
+          (completion-metadata-get metadata 'cycle-sort-function))
+      metadata
+    `(metadata
+      (display-sort-function . prescient--post-sort)
+      (cycle-sort-function . prescient--post-sort)
+      ,@(cdr metadata))))
+
+;;;;; Filtering functions
+
+(defun prescient--completion-style-filter (string table pred)
```

Comment:
Here you define a new filter function. Wouldn't it make sense to rewrite prescient-filter to use all-completions and use that?